core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -31,14 +31,18 @@ import org.apache.spark.util.Utils
  * old stages to be resubmitted, such as shuffle map fetch failures.
  */
 @DeveloperApi
-sealed trait TaskEndReason
+sealed trait TaskEndReason {
+  def shouldTaskEndEventuallyFailJob: Boolean
```

Comment:
actually, should this be a property of `TaskFailedReason`? This property doesn't make sense for `Success`, for instance. I would also make it default to `true` there